numpy/polynomial/_polybase.py

```diff
@@ -303,12 +316,25 @@ def __init__(self, coef, domain=None, window=None):
                 raise ValueError("Window has wrong number of elements.")
             self.window = window
 
+        # Validation for symbol
+        if not isinstance(symbol, str):
+            raise TypeError("Symbol must be a non-empty string")
+
+        if len(symbol) < 1:
+            raise ValueError("Symbol must be a non-empty string")
+
+        if symbol[0].isnumeric():
+            raise ValueError("First character of symbol must be non-numeric")
```

Comment:
This allows "symbols" such as `symbol='a   b   '` and `symbol='x + y / z'`.  Is that OK?  Perhaps symbols should be limited to identifiers. (With [PEP 3131](https://peps.python.org/pep-3131/), Unicode identifiers are allowed, so one could still use, for example, `symbol='λ'`.)  In that case, this code could be simplified to
```suggestion
        if not symbol.isidentifier():
            raise ValueError("Symbol string must be a valid Python identifier")
```

Reply:
> This allows "symbols" such as symbol='a b ', symbol='x + y / z' and even symbol=' '. Is that OK?

I don't have a strong opinion about this. In principle, there may be a use-case where someone wants to use an expression to represent the indeterminate (e.g. `symbol="x+1"`), but that case could also be covered with a symbol substitution (e.g. `u = x+1`).

I'm inclined to agree with this suggestion - if it turns out there is loud user demand for multi-character symbols we can always revisit. Thanks @WarrenWeckesser 

PS - do you have any other suggestions related to the overall proposal in this PR? I think this has been stuck in the "needs decision" phase for a while so if there is a sense that it's not useful I'd be inclined to just close it.

Reply:
I made the suggested change (slightly modified to account for exception handling) in 6cbde0f - LMK what you think!